File: Day_3/Part_1.py
with open('Day_3/input.txt', 'r') as file:
    data = file.readlines()


def count_trees(area, x_step, y_step):
    x = y = trees = 0
    while y < len(area):
        if area[y][x] == '#':
            trees -= -1
        x, y = (x + x_step) % (len(area[y]) - 1), y + y_step
    return trees


print(count_trees(data, 3, 1))

File: Day_3/Part_1.c
#include <stdio.h>
#include <stdlib.h>

int count_trees(char area[323][64], int x_step, int y_step);

int main()
{
    FILE *file;
    file = fopen("input.txt", "r");

    //read file into array
    char area[323][64];
    int i;

    if (file == NULL)
    {
        printf("Error Reading File\n");
        exit(0);
    }

    for (i = 0; i < 323; i++)
    {
        fscanf(file, "%s", area[i]);
    }

    int num;

    num = count_trees(area, 3, 1);

    printf("%d\n", num);
}

int count_trees(char area[323][64], int x_step, int y_step)
{
    int num = 0;
    int mod = 31;
    int x = 0;
    int y = 0;
    for (int i = 0; i < 323; i++)
    {
        if (area[y][x] == '#')
        {
            num++;
        }
        x = (x + x_step) % mod;
        y = y + y_step;
    }
    return num;
}